### ISSUE TITLE
how to get the calibration.txt

### ISSUE THREAD

**Issue by vigneshgig** [gengshan-y/high-res-stereo#5]
I have some doubt
what is doff,ndisp ?
what is unit of baseline?
if possible can give me what library is been used for camera calibration any link.
and also whether I have to send  a unrectify image or stereo rectified image using cv2.stereorectify.

unrectified image
![image](https://user-images.githubusercontent.com/34392627/62465473-9430e100-b7ac-11e9-8181-2b252bde9867.png)

rectified image
![image](https://user-images.githubusercontent.com/34392627/62465479-9a26c200-b7ac-11e9-96f4-6f8be02ebf1e.png)

Thanks

**Comment by peachthiefmedia** [comment]
I don't know how much I can help but the calibration text looks to be based on the middlebury calibration input (http://vision.middlebury.edu/stereo/data/scenes2014/ shown at the bottom)

```
Explanation:

cam0,1:        camera matrices for the rectified views, in the form [f 0 cx; 0 f cy; 0 0 1], where
  f:           focal length in pixels
  cx, cy:      principal point  (note that cx differs between view 0 and 1)

doffs:         x-difference of principal points, doffs = cx1 - cx0

baseline:      camera baseline in mm

width, height: image size

ndisp:         a conservative bound on the number of disparity levels;
               the stereo algorithm MAY utilize this bound and search from d = 0 .. ndisp-1

isint:         whether the GT disparites only have integer precision (true for the older datasets;
               in this case submitted floating-point disparities are rounded to ints before evaluating)

vmin, vmax:    a tight bound on minimum and maximum disparities, used for color visualization;
               the stereo algorithm MAY NOT utilize this information

dyavg, dymax:  average and maximum absolute y-disparities, providing an indication of
               the calibration error present in the imperfect datasets.
```

**Comment by vigneshgig** [comment]
thanks for the reply. I figured out that
but i'm getting a bad output with pretrained model but I changed the calibration file according to my camera.
input 


![left_1](https://user-images.githubusercontent.com/34392627/62618115-0e8a6e00-b931-11e9-9909-d3cc374d3f68.png)
![right_1](https://user-images.githubusercontent.com/34392627/62618117-0f230480-b931-11e9-9164-b99119167f0c.png)


output
![testing-disp](https://user-images.githubusercontent.com/34392627/62618036-e438b080-b930-11e9-9984-0c7a8aec2990.png)
![testing-ent](https://user-images.githubusercontent.com/34392627/62618037-e438b080-b930-11e9-8c8c-28ec994bbce7.png)

and also I tried rectified image I am geting same result. 
I think pretrained model does not support other domain dataset.

any suggestion

thanks

**Comment by gengshan-y** [comment]
Did you try other combinations of hyper parameters? For example, disable —clean, use larger —max_diap, and change —testres to upscale the input to around 5 mega-pixel. All the hyper parameters can be found in submission.py line 19-35.

**Comment by vigneshgig** [comment]
hey how can i convert disperity to real world depth estimation ,what clean argument is doing and also what is the unit of  max_disp. cm or m

**Comment by gengshan-y** [comment]
You can use the formulation of stereo triangulation: depth = focal_length * baseline / disparity, where focal length/disparity should be in pixel unit, and baseline/depth should be in the same unit. Focal length and baseline depend on the camera intrinsic and extrinsic. 

The "clean" argument removes depth of pixels with large uncertainty produced by the softmax layer.
